tempesta_fw/cache.c

```diff
@@ -190,6 +191,58 @@ tfw_cache_msg_cacheable(TfwHttpReq *req)
 	return __cache_method_test(req->method);
 }
 
+/*
+ * Match the given request against strings specified in "location",
+ * "cache_bypass", and "cache_fulfil" directives and according to
+ * match operators specified in the directives.
+ *
+ * Return the constant that identifies the action on a request.
+ * Currently the actions are "bypass", "fulfill", or "default".
+ */
+tfw_stmt_t
+tfw_cache_action(TfwHttpReq *req)
+{
+	TfwCfgLocation *loc;
+	TfwCfgCacheMatch *cam;
+
+	loc = tfw_location_match(&req->uri_path);
+	cam = tfw_camatch_match(loc, &req->uri_path);
+	if (cam)
+		return cam->stmt;
+	return TFW_D_CACHE_DEFAULT;
```

Comment:
There is no sense to treat default action as 3rd case - just return `TFW_D_CACHE_BYPASS` by default.